### PR TITLE
Game Improvements

### PR DESCRIPTION
**Pull request by liamjtaylor**
I have updated the word library with 58000 words from the English Dictionary. I have also added some new features to allow the word to be shown at the end if the player loses. Added error messaging when zero letters are imputed. Also the word length now randomly varies between 6 and 15 characters long. Added word length and a preview of what the word will be at entry point of the game.

Cheers

### CODE REVIEW

File: src/hangman/game.clj [hunk 1]
```diff
@@ -16,16 +15,18 @@
...
-        (do
-          (println "Input too log")
-          (get-letter))
-        (let [char (nth user-input 0)]
-          (if (contains? valid-chars char)
-            char
-            (do
-              (println "Not a valid choice")
-              (get-letter))))))))
+      (if (or (> (count user-input) 1) (< (count user-input) 1))
```
Comment (AndreaCrotti) [inline src/hangman/game.clj:18]: here we could do then (if-not (= (count user-input) 1) which would be simpler..

File: src/hangman/secret.clj [hunk 2]
```diff
@@ -33,11 +33,21 @@
   [secret]
   (str/join (map #(if (:visible %) (:char %) \_) secret)))
 
+(defn secret-string-initial
```
Comment (AndreaCrotti) [inline src/hangman/secret.clj:36]: This function is almost the same as the one above, and the docstring is also copied..
Maybe better to add a flag to the previous function?

**Comment by AndreaCrotti** [comment]
Very nice work thanks, I left some comments.

About the dictionary from the PR I can see
6,307 additions, 71,595 deletions not shown
So it looks like this dictionary you used is a lot smaller, is that right @liamjtaylor ?

**Comment by liamjtaylor** [comment]
Hi Andrea

British File: I have just reduced the dictionary to the official dictionary
and taking out words that include " 's " as this is not really good in
hangman and peoples names, the 6000 additions are because we must have had
a lot of words that matched.

Initial secret: I repeated the previous function to complete the word so if
the player lost the word was completed and the person could see what they
actually were trying to guess

Agree that the if not statement would be simpler, am new to clojure so am
just working through things with the knowledge I have.

Liam

Liam Taylor | uSwitch.com

liam.taylor@uswitch.com

Notcutt House, 36 Southwark Bridge Road, London, SE1 9EU

On 27 Apr 2015 12:08 pm, "Andrea Crotti" notifications@github.com wrote:

> Very nice work thanks, I left some comments.
> 
> About the dictionary from the PR I can see
> 6,307 additions, 71,595 deletions not shown
> So it looks like this dictionary you used is a lot smaller, is that right
> @liamjtaylor https://github.com/liamjtaylor ?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/AndreaCrotti/hangman/pull/1#issuecomment-96610977.

## 

_uSwitch is a trading name of uSwitch Ltd. Registered in *_England and 
Wales *_(Company No. 03612689). Registered Address: *_Notcutt House, 36 
Southwark Bridge Road, London, SE1 9EU*

_This communication and any attachments contains information which is 
confidential and may be subject to legal privilege. It is for intended 
recipients only. If you are not the intended recipient you must not copy, 
distribute, publish, rely on or otherwise use it without our consent. Some 
of our communications may contain confidential information which it could 
be a criminal offence for you to disclose or use without authority. If you 
have received this email in error please notify the sender immediately and 
delete the email from your computer._

_uSwitch Ltd reserves the right to monitor all email communications for 
compliance with legal, regulatory and professional standards._

**Comment by ghost** [comment]
Liam

Can you let me know how I unsubscribe from this mailing list?

Thanks

Gerald

**Comment by AndreaCrotti** [comment]
I've removed you from the collaborators now.
If that's still not enough just have a look here
https://github.com/AndreaCrotti/hangman/pull/1#issuecomment-96618839

and click on unsubscribe..

2015-04-27 13:38 GMT+01:00 gvim notifications@github.com:

**Comment by AndreaCrotti** [comment]
Ok thanks @liamjtaylor I've merged it now, I'll see if I can improve it a bit later..

If you want to give a go to the graphics (with quill) or the user authentication feel free to try something out these are the next things I wanted to do.
I also want to encapsulate the various services with the components library which might also be interesting.

**Pull request merged**